openscenario/openscenario_interpreter/src/syntax/init_actions.cpp

```diff
@@ -87,20 +91,28 @@ auto InitActions::startInstantaneousActions() -> void
   for (auto && e : global_actions) {
     e.as<GlobalAction>().start();
   }
-  for (auto && e : user_defined_actions) {
-    auto & user_defined_action = e.as<UserDefinedAction>();
-    if (user_defined_action.endsImmediately()) {
-      user_defined_action.start();
+
+  for (size_t index{0}; auto && e : user_defined_actions) {
```

Comment:
This repository is compiled with C++17. Don't take advantage of C++20 features yet.
```suggestion
    std::size_t index = 0;
    for (auto && user_defined_action : user_defined_actions) {
```